pkg/bootstrap/platform/gcp.go

```diff
@@ -156,64 +178,28 @@ func (e *gcpEnv) shouldFillMetadata() bool {
 // Metadata returns GCP environmental data, including project, cluster name, and
 // location information.
 func (e *gcpEnv) Metadata() map[string]string {
-	md := map[string]string{}
-	if e == nil {
-		return md
-	}
-	if GCPMetadata == "" && !e.shouldFillMetadata() {
-		return md
+	// If we cannot reach the metadata server, bail out with only statically defined metadata
+	fillMetadata := e.shouldFillMetadata()
+	if !fillMetadata {
+		return GCPStaticMetadata
 	}
 
 	e.Lock()
 	defer e.Unlock()
+	// Use previously computed result...
 	if e.metadata != nil {
 		return e.metadata
 	}
-	envPid, envNPid, envCN, envLoc := parseGCPMetadata()
 
-	if envPid != "" {
-		md[GCPProject] = envPid
-	}
-	if envNPid != "" {
-		md[GCPProjectNumber] = envNPid
-	}
-	if envLoc != "" {
-		md[GCPLocation] = envLoc
-	}
-	if envCN != "" {
-		md[GCPCluster] = envCN
-	}
-
-	if e.shouldFillMetadata() {
-		// suppliers is an array of functions that supply the metadata for missing properties
-		var suppliers []metadataSupplier
-		if _, found := md[GCPProject]; !found {
-			suppliers = append(suppliers, createMetadataSupplier(GCPProject, projectIDFn))
-		}
-		if _, found := md[GCPProjectNumber]; !found {
-			suppliers = append(suppliers, createMetadataSupplier(GCPProjectNumber, numericProjectIDFn))
-		}
-		if _, found := md[GCPLocation]; !found {
-			suppliers = append(suppliers, createMetadataSupplier(GCPLocation, clusterLocationFn))
-		}
-		if _, found := md[GCPCluster]; !found {
-			suppliers = append(suppliers, createMetadataSupplier(GCPCluster, clusterNameFn))
-		}
-
-		wg := waitForMetadataSuppliers(suppliers, md)
-		wg.Wait()
-	}
-
-	if GCPQuotaProjectVar != "" {
-		md[GCPQuotaProject] = GCPQuotaProjectVar
-	}
-	// Exit early now if not on GCE. This allows setting env var when not on GCE.
-	if !e.shouldFillMetadata() {
-		e.metadata = md
-		return md
-	}
+	// Start off with static metadata (if there is any)
+	md := maps.Clone(GCPStaticMetadata)
 
+	// suppliers is an array of functions that supply the metadata for missing properties
 	suppliers := []metadataSupplier{
+		createMetadataSupplier(GCPProject, projectIDFn),
```

Comment:
we should check these fields first from GCP_METADATA and if it is not available then we should rely on compute metadata server ?

Reply:
Well we do get them from GCP_METADATA. but then possible override them. yeah I guess we should probably prefer the env, will update it. good idea

Reply:
This is already handled actually, I forgot: https://github.com/istio/istio/blob/ac876ff08c16859048e3b2d959ca9c09e8de3b5a/pkg/bootstrap/platform/gcp.go#L226